git-try-push/main.js

```diff
@@ -46,8 +46,11 @@ async function main() {
         for (let i = 0; i < tries; i++) {
             try {
                 // Try to push, if successful, then checkout previous branch and just exit.
-                // Don't try to force push the first time in case it's not necessary.
-                if (force && i>0)
+				// If force pushing with lease, don't try to force push the first time
+				// in case it's not necessary.
+				// If force pushing without lease, force push the first time since we've
```

Comment:
IMO it's still probably worth attempting first time to not force push in case it's not necessary?

Reply:
Our publish workflow currently only ever requests `--force` (without lease) whenever it is necessary.

If we want to make sure to try a plain `git push` first, then the `pull --rebase` step below needs to be modified/skipped, because it  destroys our autosquashed changes. 

Would you rather we did that?

Reply:
> > Our publish workflow currently only ever requests --force (without lease) whenever it is necessary.
> 
> 
> 
> This action is not used exclusively by that workflow 😁 

I'm aware! That's why this behaviour is gated by `no_lease`, which is false(-y) by default.

Reply:
But, thinking it through:

If `git push` would've worked without `--force`, then `--force` does no harm (right?).

If `git push` didn't work at first, then skipping/modifying `pull --rebase` will result in the same harm (if any!) that doing `--force` does on the second go.

So not sure there's much value in trying to be more careful here. But feel free to let me know if I'm missing something here!